### PR TITLE
Simplify gem usage

### PR DESCRIPTION
**Pull request by mohameddiaa27**
## Description

Improves the gem interface

## PR Contains:

- [ ] Documentation
- [ ] Tests
- [x] Breaking Changes

## Related Issues

- resolves https://github.com/catawiki/rate-limit/issues/23

### Added

- added `Worker#raise_errors`
- added `Worker#only_failures`

### Deprecated

- `RateLimit.throttle_with_block!`
- `RateLimit.throttle_only_failures_with_block!`

### Removed

- removed `Worker#throttle_with_block!`
- removed `Worker#throttle_only_failures_with_block!`

### CODE REVIEW

**Comment by dhruvCW** [comment]
missing docs update

**Comment by mohameddiaa27** [comment]
> missing docs update

Here it is https://github.com/catawiki/rate-limit/pull/24

File: lib/rate_limit/throttler.rb [hunk 1]
```diff
@@ -5,26 +5,16 @@ module Throttler
...
-    def throttle_with_block!
-      failure!(raise_errors: true) if reloaded_limit_exceeded?
+      unless only_failures
+        yield if block_given?
 
-      yield
+        return success!
+      end
 
+      yield if block_given?
```
Comment (dhruvCW) [inline lib/rate_limit/throttler.rb:14]: could we written as 

```ruby
yield if block_given?

return success! unless only_failures
```

wdyt ?

**Review by dhruvCW — APPROVED** [review]
one small suggestion to avoid duplicate code, but looks great otherwise 👍

File: lib/rate_limit/throttler.rb (hunk 1 above)
Comment (mohameddiaa27) [reply lib/rate_limit/throttler.rb]: Makes sense 💯  changed that  👍

File: lib/rate_limit/throttler.rb [hunk 2]
```diff
@@ -5,26 +5,14 @@ module Throttler
...
-      failure!(raise_errors: true) if reloaded_limit_exceeded?
-
-      yield
+      yield if block_given?
 
+      return success! unless only_failures
+    rescue Errors::LimitExceededError => e
+      raise e
+    rescue StandardError => e
       success!
```
Comment (dhruvCW) [inline lib/rate_limit/throttler.rb:14]:
```suggestion
    rescue StandardError => e
      success! unless e.is_a?(Errors::LimitExceededError)
```

**Review by dhruvCW — APPROVED** [review]

**Pull request merged**